test/src/test/java/org/corfudb/integration/LogReplicationDynamicStreamIT.java

```diff
@@ -0,0 +1,520 @@
+        // The plus one is for the clear entry before applying the stream updates
+        int expectedMessageSize = numWrites + 1;
+        int numTables = 1;
+        CountDownLatch streamingStandbySnapshotCompletion = new CountDownLatch(expectedMessageSize);
+        CountDownLatch numTxLatch = new CountDownLatch(numTables);
+        // Subscribe the testing stream listener
+        StreamingStandbyListener listener = new StreamingStandbyListener(streamingStandbySnapshotCompletion,
+                numTxLatch, Collections.singleton(streamId));
+        corfuStoreStandby.subscribeListener(listener, NAMESPACE, DefaultLogReplicationConfigAdapter.TAG_ONE,
+                CorfuStoreMetadata.Timestamp.newBuilder().setEpoch(0).setSequence(0).build());
+
+        // Verify the stream listener received all the updates.
+        streamingStandbySnapshotCompletion.await();
+        numTxLatch.await();
+        assertThat(listener.messages.size()).isEqualTo(expectedMessageSize);
+
+        // Verify snapshot sync succeeded and stream listener received all the changes
+        verifyDataOnStandby(mapTagOneStandby, numWrites);
+    }
+
+    /**
+     * This test will verify the streams to stream tags map is correctly rebuilt during log entry sync.
+     *
+     * (1) Perform a basic snapshot sync and log entry sync, verify the cluster is in log entry sync state
+     * (2) Open a new map only on Source side with TAG_ONE
+     * (3) Write some new entries to this new map on Source
+     * (4) Initiate a testing stream listener and subscribe to TAG_ONE on Sink from timestamp 0
+     * (5) Verify new data get replicated and the stream listener received all the updates
+     */
+    @Test
+    public void testStandbyStreamingLogEntrySyncNewTable() throws Exception {
+        // perform basic snapshot and log entry sync, the cluster should be in log entry sync state now
+        testSnapshotAndLogEntrySync();
+
+        // Verify log replication status is log entry sync
+        verifyInLogEntrySyncState();
+
+        // Open testing map on Source and Sink
+        String streamName = "TableStreaming001";
+
+        Table<StringKey, IntValue, Metadata> mapTagOne = corfuStoreActive.openTable(
+                NAMESPACE,
+                streamName,
+                StringKey.class,
+                IntValue.class,
+                Metadata.class,
+                TableOptions.fromProtoSchema(ValueFieldTagOne.class)
+        );
+        writeToMap(mapTagOne, true, 0, numWrites);
+
+        // Make sure Sink side record comes from log entry sync instead of opening table by itself.
+        while (!standbyRuntime.getTableRegistry().getRegistryTable().containsKey(
+                TableName.newBuilder().setTableName(streamName).setNamespace(NAMESPACE).build())) {
+            log.trace("Test table record hasn't been replicated");
+        }
+
+        // Register the table into table map, which is required before subscribe the listener.
```

Comment:
can you elaborate a bit more on this comment?  What does 'register' mean?  Does it mean the in-memory variables, tables etc in TableRegistry need to be populated with this table's info?

Reply:
Modified a little bit, it means the table needs to be opened through "corfustore.openTable()" before creating streaming tasks.
https://github.com/CorfuDB/CorfuDB/blob/762ecd8cdf102f9dc0595cf2a8f6f794a36d6e8b/runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java#L621